lib/lightstep/tracer.rb

```diff
@@ -213,5 +220,27 @@ def extract_from_text_map(operation_name, carrier)
 
       span
     end
+
+    def inject_to_http_headers(span, carrier)
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'spanid'] = span.span_context.id
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'traceid'] = span.span_context.trace_id unless span.span_context.trace_id.nil?
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'sampled'] = 'true'
+
+      span.span_context.baggage.each do |key, value|
+        carrier[CARRIER_BAGGAGE_PREFIX + key.gsub(/[^A-Za-z0-9-]/, '')] = value
+      end
+    end
+
+    def extract_from_rack_env(operation_name, env)
+      extract_from_text_map(operation_name, env.reduce({}){|memo, tuple|
+        raw_header, value = tuple
+        header = raw_header.gsub(/^HTTP_/, '').gsub("_","-").downcase
```

Comment:
There's no reason a baggage item can't use `"_"`, and this will transform it... not sure how to be correct in the presence of Rack here. :(  But we could have this use a regexp to match the tracer state XOR baggage prefix, then just chop off the rest of the string and canonicalize the suffix as needed.